### PR TITLE
Refactor instruction processor to accept SVMInstruction

### LINKED ISSUE #2112: process_compute_budget_instructions use non-owned `SVMInstruction` from svm-transaction
### Problem

- SVM calls `process_compute_budget_instructions`
- `SVMTransaction` only has non-owning `SVMInstruction` instead of `CompiledInstruction`
- `process_compute_budget_instructions` currently forces use of owned `CompiledIntstruction` 

### Solution

- Make `process_compute_budget_instructions` accept `SVMInstruction`

### PR DESCRIPTION
**Pull request by tao-stones**
#### Problem

#2112 

#### Summary of Changes
- update `runtime-transaction::process_compute_budget_instructions`, and downstream functions to accept `SVMInstruction`.
- update call sites to supply `SVMInstruction` iter

Fixes #2112

### CODE REVIEW

File: core/src/banking_stage/immutable_deserialized_packet.rs [hunk 1]
```diff
@@ -64,7 +65,8 @@ impl ImmutableDeserializedPacket {
         } = process_compute_budget_instructions(
             sanitized_transaction
                 .get_message()
-                .program_instructions_iter(),
+                .program_instructions_iter()
+                .map(|(pubkey, ix)| (pubkey, SVMInstruction::from(ix))),
```
Comment (tao-stones) [inline core/src/banking_stage/immutable_deserialized_packet.rs:69]: it's `SanitizedVersionedMessage` that doesn't impl `SVMMessage` trait, doing `map` directly.

**Review by apfitzge — COMMENTED** [review]
Changes look good to me.

I think most of these are correct but we could probably get some more "clean" syntax in several places.

In several places you are explicitly calling `SVMMessage::program_instructions_iter`. Once those get changed over to accepting an `&impl SVMMessage` we'd want to just change those calls back to `{message,tx}.program_instructions_iter()`. It might be best to just change the argument now, wdyt?

File: cost-model/src/cost_model.rs [hunk 2]
```diff
@@ -151,7 +151,7 @@ impl CostModel {
 
     fn get_transaction_cost(
         tx_cost: &mut UsageCostDetails,
-        transaction: &SanitizedTransaction,
+        transaction: &impl SVMTransaction,
```
Comment (apfitzge) [inline cost-model/src/cost_model.rs]: small nit - I think we technically only need an `SVMMessage` here. We don't use the additional signature accessors from `SVMTransaction` afaict

**Comment by tao-stones** [comment]
> Changes look good to me.
> 
> I think most of these are correct but we could probably get some more "clean" syntax in several places.
> 
> In several places you are explicitly calling `SVMMessage::program_instructions_iter`. Once those get changed over to accepting an `&impl SVMMessage` we'd want to just change those calls back to `{message,tx}.program_instructions_iter()`. It might be best to just change the argument now, wdyt?

thought to leave these for separate refactor, but small change fit in commit 3 :)

File: cost-model/src/cost_model.rs (hunk 2 above)
Comment (tao-stones) [reply cost-model/src/cost_model.rs]: 5640a49ba1b6a0c39c3133b19abc953e6fbdadd6

**Review by apfitzge — APPROVED** [review]

**Pull request merged**